include/deal.II/lac/trilinos_precondition.h

```diff
@@ -113,6 +114,22 @@ namespace TrilinosWrappers
     void clear ();
 
     /**
+     * Return the MPI communicator object in use with this matrix.
+     */
+    MPI_Comm get_mpi_communicator () const;
```

Comment:
All of the changes to this file seem necessary, but ultimately independent of the overall PR. If you feel like it, feel free to break this out into a PR of its own (it also deserves a changelog entry of its own).